presto-hive-functions/src/main/java/com/facebook/presto/hive/functions/HiveFunctionNamespaceManager.java

```diff
@@ -0,0 +1,284 @@
+        return function.getFunctionMetadata();
+    }
+
+    public ScalarFunctionImplementation getScalarFunctionImplementation(FunctionHandle functionHandle)
+    {
+        checkArgument(functionHandle instanceof HiveFunctionHandle);
+        HiveFunction function = functions.getUnchecked(FunctionKey.from((HiveFunctionHandle) functionHandle));
+        verify(function instanceof HiveScalarFunction);
+        return ((HiveScalarFunction) function).getJavaScalarFunctionImplementation();
+    }
+
+    @Override
+    public CompletableFuture<SqlFunctionResult> executeFunction(String source, FunctionHandle functionHandle, Page input, List<Integer> channels, TypeManager typeManager)
+    {
+        return null;
+    }
+
+    @Override
+    public void addUserDefinedType(UserDefinedType userDefinedType)
+    {
+        return;
+    }
+
+    @Override
+    public Optional<UserDefinedType> getUserDefinedType(QualifiedObjectName typeName)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public boolean canResolveFunction()
+    {
+        return true;
+    }
+
+    private HiveFunction createFunction(FunctionKey key)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
+            QualifiedObjectName name = key.getName();
+            try {
+                Class<?> functionClass = hiveFunctionRegistry.getClass(name);
+                if (anyAssignableFrom(functionClass, GenericUDF.class, UDF.class)) {
+                    return HiveScalarFunction.create(functionClass, name, key.getArgumentTypes(), typeManager);
+                }
+                else {
+                    throw unsupportedFunctionType(functionClass);
+                }
+            }
+            catch (ClassNotFoundException e) {
+                throw functionNotFound(name.toString());
+            }
+        }
+    }
+
+    private static boolean anyAssignableFrom(Class<?> cls, Class<?>... supers)
+    {
+        return Stream.of(supers).anyMatch(s -> s.isAssignableFrom(cls));
+    }
+
+    private static class EmptyTransactionHandle
+            implements FunctionNamespaceTransactionHandle
+    {
+    }
+
+    private static class FunctionKey
+    {
+        private final QualifiedObjectName name;
+        private final List<TypeSignature> argumentTypes;
+
+        public static FunctionKey from(HiveFunctionHandle handle)
+        {
+            final Signature signature = handle.getSignature();
+            return FunctionKey.of(signature.getName(), signature.getArgumentTypes());
+        }
+
+        private static FunctionKey of(QualifiedObjectName name, List<TypeSignature> argumentTypes)
+        {
+            return new FunctionKey(name, argumentTypes);
+        }
+
+        private FunctionKey(QualifiedObjectName name, List<TypeSignature> argumentTypes)
+        {
+            this.name = name;
```

Comment:
`requireNonNull`.  Generally speaking always check non null in constructors, unless explicitly expect null, then annotate with `@Nullable`.